account_avatax_oca/models/account_move.py

```diff
@@ -252,26 +251,42 @@ def _avatax_compute_tax(self, commit=False):
                     tax = Tax.get_avalara_tax(rate, doc_type)
                     if tax and tax not in line.tax_ids:
                         line_taxes = line.tax_ids.filtered(lambda x: not x.is_avatax)
-                        taxes_to_set.append((index, line_taxes | tax))
+                        taxes_to_set[line.id] = line_taxes | tax
                     line.avatax_amt_line = tax_result_line["tax"]
             self.with_context(check_move_validity=False).avatax_amount = tax_result[
                 "totalTax"
             ]
             container = {"records": self}
-            self.with_context(
-                avatax_invoice=self, check_move_validity=False
-            )._sync_dynamic_lines(container)
-            self.line_ids.mapped("move_id")._check_balanced(container)
 
             # Set Taxes on lines in a way that properly triggers onchanges
             # This same approach is also used by the official account_taxcloud connector
-            with Form(self) as move_form:
-                for index, taxes in taxes_to_set:
-                    with move_form.invoice_line_ids.edit(index) as line_form:
-                        line_form.tax_ids.clear()
-                        for tax in taxes:
-                            line_form.tax_ids.add(tax)
-
+            with self.with_context(
+                avatax_invoice=self, check_move_validity=False
+            )._sync_dynamic_lines(container), self.line_ids.mapped(
+                "move_id"
+            )._check_balanced(
+                container
+            ):
+                for line_id in taxes_to_set.keys():
+                    line = self.invoice_line_ids.filtered(lambda x: x.id == line_id)
+                    line.tax_ids.write({"tax_ids": [(6, 0, [])]})
```

Comment:
Raising error, I'm not sure which it's the purpose, but I suggest remove this line or change it for line.write({"tax_ids": [(6, 0, [])]})